ofrak_patch_maker/ofrak_patch_maker/toolchain/gnu.py

```diff
@@ -371,7 +371,7 @@ def __init__(
         )
 
         if self._config.relocatable:
-            self._compiler_flags.append("-pie")
+            self._compiler_flags.append("-fPIE")
             self._linker_flags.append("--pic-executable")
```

Comment:
Hard to tell from the docs, but it looks like the compiler wants `-fPIE` if the linker will be using `-pie`. Documentation for `ld` indicates that `--pic-executable` is equivalent to `-pie`, but documentation for GCC omits any mention of `--pic-executable`... I wonder if we should be setting `pie` instead of `--pic-executable`?

- <https://gcc.gnu.org/onlinedocs/gcc/Code-Gen-Options.html#index-fpie>
- <https://gcc.gnu.org/onlinedocs/gcc/Link-Options.html#index-pie>
- <https://man7.org/linux/man-pages/man1/ld.1.html>

Reply:
As far as I can tell, `--pic-executable` is just the long-form alias for `-pie`. If `-pie` seems more readable I'll change it otherwise I'll leave it as-is.

Reply:
Changed this to `-pie` in case we decide to use gcc for linking in the future (gcc only supports `pie` not `pic-executable`)